scalameta/transversers/shared/src/main/scala/scala/meta/transversers/SimpleTraverser.scala

```diff
@@ -0,0 +1,6 @@
+package scala.meta
+package transversers
+
+private[meta] class SimpleTraverser {
+  def apply(tree: Tree): Unit = tree.children.foreach(apply)
```

Comment:
How about we get rid of the autogenerated Traverser altogether? Then SimpleTraverser would simply become Traverser.

Reply:
done

Reply:
this causes Mima to complain because the macro used to generate some additional `apply` methods which are no longer needed:
```
[error]  * method apply(scala.Option)
[error]  * method apply(scala.collection.immutable.List)
[error]  * method apply(scala.Option,org.scalameta.overload.Hack1)
[error]  * method apply(scala.collection.immutable.List,org.scalameta.overload.Hack2)
[error]  * object scala.meta.transversers.Traverser does not have a correspondent in current version
```
Are you ok in adding exceptions in Mima for those or maybe have a better suggestion since it breaks backwards compatibility?